src/main/java/org/isf/medicalinventory/gui/InventoryEdit.java

```diff
@@ -315,91 +429,307 @@ private JPanel getPanelContent() {
 	private JPanel getPanelFooter() {
 		if (panelFooter == null) {
 			panelFooter = new JPanel();
-			panelFooter.add(getNewButton());
+			panelFooter.add(getSaveButton());
 			panelFooter.add(getDeleteButton());
+			panelFooter.add(getLotButton());
+			panelFooter.add(getCleanTableButton());
 			panelFooter.add(getCloseButton());
 		}
 		return panelFooter;
 	}
 
 	private GoodDateChooser getJCalendarFrom() {
 		if (jCalendarInventory == null) {
-			jCalendarInventory = new GoodDateChooser(LocalDate.now());
+			jCalendarInventory = new GoodDateChooser(LocalDate.now(), false, false);
 			if (inventory != null) {
 				jCalendarInventory.setDate(inventory.getInventoryDate().toLocalDate());
+				dateInventory = inventory.getInventoryDate();
 			}
 			jCalendarInventory.addDateChangeListener(event -> {
-				dateInventory = jCalendarInventory.getDateStartOfDay();
+				dateInventory = jCalendarInventory.getDate().atStartOfDay();
 			});
 		}
 		return jCalendarInventory;
 	}
 
-	private JButton getNewButton() {
+	private JButton getSaveButton() {
 		saveButton = new JButton(MessageBundle.getMessage("angal.common.save.btn"));
 		saveButton.setMnemonic(MessageBundle.getMnemonic("angal.common.save.btn.key"));
 		saveButton.addActionListener(actionEvent -> {
 			String State = InventoryStatus.draft.toString();
```

Comment:
Please refactor this

```suggestion
			String state = InventoryStatus.draft.toString();
```